### PR TITLE
Prevent docs gen workflow on forks

### LINKED ISSUE #6386: [CT-1609] [Bug] `Generate CLI API docs` CI check not able to checkout PR branches from forks
### Is this a new bug in dbt-core?

- [X] I believe this is a new bug in dbt-core
- [X] I have searched the existing issues, and I could not find an existing issue for this bug

### Current Behavior

When reviewing a community contributed PR, the **Generate CLI API docs** CI check failed with this message:

>   Error: A branch or tag with the name 'feature/dbt-deps-tarball' could not be found

Instance of this issue:
https://github.com/dbt-labs/dbt-core/actions/runs/3625104874/jobs/6113159381

### Expected Behavior

Hoped that the CI check could run without erroring prematurely. e.g., if needed, checkout the branch from a fork rather than a local branch.

### Steps To Reproduce

1. Run to run CI on a PR from a fork (i.e., a PR submitted by someone outside the dbt-labs organization)

### Relevant log output

```shell
Run actions/checkout@v3
Syncing repository: dbt-labs/dbt-core
Getting Git version info
Temporarily overriding HOME='/home/runner/work/_temp/1d57006c-8385-4aa4-92f0-1c31e45309b6' before making global git config changes
Adding repository directory to the temporary git global config as a safe directory
/usr/bin/git config --global --add safe.directory /home/runner/work/dbt-core/dbt-core
Deleting the contents of '/home/runner/work/dbt-core/dbt-core'
Initializing the repository
Disabling automatic garbage collection
Setting up auth
Fetching the repository
Determining the checkout info
  /usr/bin/git branch --list --remote origin/feature/dbt-deps-tarball
  /usr/bin/git tag --list feature/dbt-deps-tarball
  Error: A branch or tag with the name 'feature/dbt-deps-tarball' could not be found
```

### Environment

N/A

### Which database adapter are you using with dbt?

_No response_

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by stu-k**
resolves #6386

### Description

Adds a conditional to the `check_gen` step of the CLI API doc gen workflow to prevent being run off of forks. Tested on [a fork](https://github.com/dbt-labs/dbt-core/pull/6389):
Screen Shot 2022-12-06 at 9 38 05 AM

### Checklist

- [X] I have read [the contributing guide](https://github.com/dbt-labs/dbt-core/blob/main/CONTRIBUTING.md) and understand what's expected of me
- [X] I have signed the [CLA](https://docs.getdbt.com/docs/contributor-license-agreements)
- [X] I have run this code in development and it appears to resolve the stated issue
- [X] This PR includes tests, or tests are not required/relevant for this PR
- [X] I have [opened an issue to add/update docs](https://github.com/dbt-labs/docs.getdbt.com/issues/new/choose), or docs changes are not required/relevant for this PR
- [X] I have run `changie new` to [create a changelog entry](https://github.com/dbt-labs/dbt-core/blob/main/CONTRIBUTING.md#adding-a-changelog-entry)

### CODE REVIEW

**Review by dbeatty10 — APPROVED** [review]
Awesome, @stu-k ! 💪

Thanks for the screenshot -- made it easy to see it working 🧠 

Approving this proactively assuming that all CI tests will pass.

File: .changes/unreleased/Under the Hood-20221206-094015.yaml
Comment (emmyoop) [inline .changes/unreleased/Under the Hood-20221206-094015.yaml:7]: This change doesn't need a changelog since it's not something actually released.

**Review by emmyoop — APPROVED** [review]
Looks good - just remove the changelog before merging!

**Pull request merged**
